e2e/app/sidebar/genomic-extractions-sidebar.ts

```diff
@@ -50,10 +51,13 @@ export default class GenomicExtractionsSidebar extends BaseSidebar {
     try {
       const table = this.getHistoryTable();
       await table.waitUntilVisible();
-      await this.page.waitForXPath(`${table.getXpath()}//*[@data-icon="check-circle" and @role="img"]`, {
-        visible: true,
-        timeout
-      });
+      await Promise.all([
+        this.page.waitForXPath(`${table.getXpath()}//*[@data-icon="check-circle" and @role="img"]`, {
+          visible: true,
+          timeout
+        }),
+        waitWhileLoading(this.page, { includeRuntimeSpinner: true, timeout })
+      ]);
```

Comment:
Avoid false success if there are multiple rows in Extraction table. Add checks for extraction running spinner to be certain extraction has completed.